src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -42,80 +47,138 @@ var kubeconfigCmd = &cobra.Command{
-				// For a symlink pointing elsewhere, we assume we can overwrite.
-				fmt.Println("false")
-			}
-			return nil
+		winConfig, err := readKubeConfig(winConfigPath)
+		if err != nil {
+			return err
 		}
-		if enable {
-			err = os.Mkdir(configDir, 0o750)
-			if err != nil && !errors.Is(err, os.ErrExist) {
-				// The error already contains the full path, we can't do better.
-				return err
-			}
-			err = os.Symlink(configPath, linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrExist) {
-					// If it already exists, do nothing; even if it's not a symlink.
-					return nil
-				}
-				return err
-			}
-		} else {
-			// No need to create if we want to remove it
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					return nil
-				}
-				return err
-			}
-			if target == configPath {
-				err = os.Remove(linkPath)
-				if err != nil && !errors.Is(err, os.ErrNotExist) {
-					return err
-				}
-			}
+
+		linuxConfigDir := path.Join(homedir.HomeDir(), ".kube")
+		linuxConfig, err := readKubeConfig(filepath.Join(linuxConfigDir, "config"))
+		if err != nil {
+			return err
+		}
+
+		cleanConfig := removeExistingRDConfig(rdCluster, &linuxConfig)
+
+		kubeConfig, err := updateClusterIP(winConfig, *cleanConfig, rdNetworking)
+
+		var finalKubeConfigFile *os.File
+		if err := os.MkdirAll(linuxConfigDir, 0o750); err != nil {
+			return err
+		}
+		finalKubeConfigFile, err = os.Create(filepath.Join(linuxConfigDir, "config"))
+		if err != nil {
+			return err
+		}
+		defer finalKubeConfigFile.Close()
+		err = os.MkdirAll(linuxConfigDir, 0o750)
+		if err != nil && !errors.Is(err, os.ErrExist) {
+			// The error already contains the full path, we can't do better.
+			return err
+		}
+		err = yaml.NewEncoder(finalKubeConfigFile).Encode(kubeConfig)
+		if err != nil {
+			return err
 		}
 		return nil
 	},
 }
 
+func readKubeConfig(configPath string) (kubeConfig, error) {
+	var config kubeConfig
+	configFile, err := os.Open(configPath)
+	if err != nil {
+		return config, err
+	}
+	defer configFile.Close()
+	err = yaml.NewDecoder(configFile).Decode(&config)
+	if err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
+
+func updateClusterIP(winConfig, linuxConfig kubeConfig, rdNetworking bool) (kubeConfig, error) {
+	ip, err := getClusterIP()
+	if err != nil {
+		return winConfig, err
+	}
+	// Fix up any clusters at 127.0.0.1, using the IP address we found.
+	for clusterIdx, cluster := range winConfig.Clusters {
+		server, err := url.Parse(cluster.Cluster.Server)
+		if err != nil {
+			// Ignore any clusters with invalid servers
+			continue
+		}
+		if winConfig.Clusters[clusterIdx].Name != rdCluster {
+			continue
+		}
+		if rdNetworking {
+			server.Host = net.JoinHostPort("gateway.rancher-desktop.internal", server.Port())
+		} else {
+			if server.Port() != "" {
+				server.Host = net.JoinHostPort(ip.String(), server.Port())
+			} else {
+				server.Host = ip.String()
+			}
+
+		}
```

Comment:
The `rdNetworking` case needs to handle empty port.  But then, at that point it's easier to deal with the port separately:
```suggestion
		host := "gateway.rancher-desktop.internal"
		if !rdNetworking {
			host = ip.String()
		}
		if server.Port() != "" {
			host = net.JoinHostPort(host, server.Port())
		}
		server.Host = host
```